### PR TITLE
[REPOSITORY] Implement a repository for the user + DTOes

### PR DESCRIPTION
**Pull request by OlexVoit**

### CODE REVIEW

File: src/main/java/mate/sep23/group3/car/sharing/dto/user/UserRegistrationRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    @NotBlank(message = NOT_BE_NULL)
+    @Length(min = 4, max = 24, message = FROM_8_TO_24_CHARACTERS)
+    private String password;
+
+    @NotBlank(message = NOT_BE_NULL)
+    @Length(min = 4, max = 24, message = FROM_8_TO_24_CHARACTERS)
+    private String repeatPassword;
+
+    @NotBlank(message = NOT_BE_NULL)
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/dto/user/UserRegistrationRequestDto.java:28]: Add @NotNull

File: src/main/java/mate/sep23/group3/car/sharing/dto/user/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private String firstName;
+
+    @NotBlank(message = NOT_BE_NULL)
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/dto/user/UserRegistrationRequestDto.java:31]: Add @NotNull

File: src/main/java/mate/sep23/group3/car/sharing/validation/FieldMatch.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Constraint(validatedBy = PasswordMatchesValidator.class)
+@Target(ElementType.TYPE)
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/validation/FieldMatch.java:11]: `ElementType.FIELD`

File: src/main/java/mate/sep23/group3/car/sharing/validation/PasswordMatchesValidator.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+import mate.sep23.group3.car.sharing.dto.user.UserRegistrationRequestDto;
+
+public class PasswordMatchesValidator implements ConstraintValidator<FieldMatch,
+        UserRegistrationRequestDto> {
+    @Override
+    public boolean isValid(UserRegistrationRequestDto user,
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/validation/PasswordMatchesValidator.java]: user -> requestDto

File: src/main/java/mate/sep23/group3/car/sharing/validation/PasswordMatchesValidator.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+                           ConstraintValidatorContext constraintValidatorContext) {
+        return user.getPassword() != null
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/validation/PasswordMatchesValidator.java:13]: `return reuestDto.getPassword().equals(requestDto.getRepeatedPassword)`

**Review by ponomvrenko — CHANGES_REQUESTED** [review]

File: src/main/java/mate/sep23/group3/car/sharing/security/JwtUtil.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    public boolean isTokenValid(String token) {
+        try {
+            Jws<Claims> claimsJwts = Jwts.parserBuilder()
+                    .setSigningKey(secret)
+                    .build()
+                    .parseClaimsJws(token);
+
+            return !claimsJwts.getBody().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            return false;
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/security/JwtUtil.java]: We should throw here `throw new JwtException("Expired or invalid JWT token!");` to notificate our user about exception type reason

**Review by ponomvrenko — CHANGES_REQUESTED** [review]

File: src/main/java/mate/sep23/group3/car/sharing/security/JwtAuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HEADER);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_START)) {
```
Comment (ponomvrenko) [inline src/main/java/mate/sep23/group3/car/sharing/security/JwtAuthenticationFilter.java]: Please change the naming of the constant to `BEARER_PREFIX`

**Review by ponomvrenko — CHANGES_REQUESTED** [review]

**Comment by ponomvrenko** [comment]
Well done! Please pay attention to my minor comments

**Review by ponomvrenko — APPROVED** [review]
Well done! LGTM

**Review by dima-semeniuk — APPROVED** [review]
Great job!

**Pull request merged**